### PR TITLE
Add ActionController::API benchmark with Kir

### PR DESCRIPTION
**Pull request by bf4**
Resolves https://github.com/ruby-bench/ruby-bench-suite/issues/57

(This is a first draft, intended to be improved)

### CODE REVIEW

**Comment by kirs** [comment]
@tgxworld check it out. We paired on this one on Railsconf

File: rails/benchmarks/bm_actioncontroller_api.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  def show
+    render json: "JSON", status: :ok
+  end
+end
+app = ApiController.action(:show)
+request = Rack::MockRequest.new(app)
+Benchmark.rails("actioncontroller/api", time: 5) do
+  response = request.get("/")
+  fail "Bad response #{response.status}" if response.status != 200
+  fail "Bad response #{response.body}" if response.body != "JSON"
```
Comment (bf4) [inline rails/benchmarks/bm_actioncontroller_api.rb:15]: Obviously there's more tests that we could add, this should be a good starting point, no?

Comment (bf4) [inline rails/benchmarks/bm_actioncontroller_api.rb:15]: Maybe add a second controller test here that using AC::Base?

**Comment by tgxworld** [comment]
Hi @bf4, sorry I missed this.

I'm just wondering if there is anything different that `ActionController::API` does?

**Comment by bf4** [comment]
@tgxworld yes, it's a smaller middleware stack and has slightly different error handling.  The smaller stack is mostly for performance reasons, removing parts not needed for an API.

File: rails/benchmarks/bm_actioncontroller_api.rb (hunk 1 above)
Comment (kirs) [inline rails/benchmarks/bm_actioncontroller_api.rb:15]: I don't think we need more tests these two.

**Comment by tgxworld** [comment]
Thanks @bf4 @kirs!

File: rails/benchmarks/bm_actioncontroller_api.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'bundler/setup'
+require 'action_controller'
+require_relative 'support/benchmark_rails'
+
+class ApiController < ActionController::API
+  def show
+    render json: "JSON", status: :ok
```
Comment (bf4) [inline rails/benchmarks/bm_actioncontroller_api.rb:7]: curiously enough, "JSON" isn't actually valid JSON (unless you're using some quirky mode processing like rails does by default). Maybe we should change this to `render plain: "TEXT", status: :ok`?

**Pull request merged**